sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfigUtil.java

```diff
@@ -27,4 +33,54 @@ static <T> T assertNotNull(@Nullable T object, String description) {
     }
     return object;
   }
+
+  /**
+   * Find a registered {@link ComponentProvider} which {@link ComponentProvider#getType()} matching
+   * {@code type}, {@link ComponentProvider#getName()} matching {@code name}, and call {@link
+   * ComponentProvider#create(StructuredConfigProperties)} with the given {@code model}.
+   *
+   * @throws ConfigurationException if no matching providers are found, or if multiple are found
+   *     (i.e. conflict), or if {@link ComponentProvider#create(StructuredConfigProperties)} throws
+   */
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  static <T> T loadComponent(SpiHelper spiHelper, Class<T> type, String name, Object model) {
+    // TODO(jack-berg): cache loaded component providers
+    List<ComponentProvider> componentProviders = spiHelper.load(ComponentProvider.class);
+    List<ComponentProvider<?>> matchedProviders =
+        componentProviders.stream()
+            .map(
+                (Function<ComponentProvider, ComponentProvider<?>>)
+                    componentProvider -> componentProvider)
```

Comment:
Is this a leftover or a workaround for java's questionable type system? 🤔

Reply:
Needed to appease the type system:
```
/Users/jberg/code/open-telemetry/opentelemetry-java/sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfigUtil.java:54: error: incompatible types: cannot infer type-variable(s) R,A,CAP#1,T#2
            .collect(Collectors.toList());
                    ^
    (argument mismatch; Collector<ComponentProvider<?>,CAP#2,List<ComponentProvider<?>>> cannot be converted to Collector<? super ComponentProvider,CAP#2,List<ComponentProvider<?>>>)
```
We can't load a parameterized SPI type, so the prior line uses the raw type `List<ComponentProvider>` instead of `List<ComponentProvider<?>>` (casting isn't allowed). So we use the `.map` to coerce the list values from being raw types to having a parameter type of `?`. Its weird. I'm open to suggestions.